app/scripts/metamask-controller.js

```diff
@@ -1619,6 +1620,19 @@ module.exports = class MetamaskController extends EventEmitter {
     return pendingNonce
   }
 
+  /**
+   * Returns the next nonce according to the nonce-tracker
+   * @param address {string} - The hex string address for the transaction
+   * @returns Promise<number>
+   */
+  async getNextNonce (address) {
+    const { nonceDetails, releaseLock} = await this.txController.nonceTracker.getNonceLock(address)
```

Comment:
Isn't `nextNonce` alongside `nonceDetails`, not inside of it?

e.g. 
```suggestion
    const { nextNonce, releaseLock} = await this.txController.nonceTracker.getNonceLock(address)
```

Reply:
Also, nitpick:

```suggestion
    const { nonceDetails, releaseLock } = await this.txController.nonceTracker.getNonceLock(address)
```